From: Djair Guilherme <djair.jr@gmail.com>
Subject: [PATCH] ov5647: add 1280x960 (2x2 binned) and 640x480 30 fps modes

The Sophgo OV5647 library only has 1920x1080 (a centre crop of the
2592x1944 array). This adds the whole array, 2x2 binned to 1296x972 (4:3),
output 1280x960 centred in it, at the same 30 fps: the window/binning
registers of the mainline Linux ov5647.c 2x2-binned mode on top of the vendor
1080p table, keeping the vendor PLL, MIPI timing, HTS 2416 and VTS 1104 (same
pixel clock, line time and exposure limits). 1280, not 1296: the ISP
post-raw stalls on a 1296-wide frame (measured: CSI overflow, no frame
done); 1280 is a multiple of 64 like 1920. Less data on MIPI and through the
ISP, and the YOLO input fills 640x480 of 640x640 instead of 640x360.

Also 640x480 @ 30 fps (mainline VGA window and binning + skipping, 0x35),
with the same vendor timing instead of the mainline 60 fps PLL.

Sensor types OV_OV5647_MIPI_1M2_30FPS_10BIT (PIC_1280x960, new) and
OV_OV5647_MIPI_480P_30FPS_10BIT (PIC_640x480) in the sample common code.
F7, docs/multimedia.md.

diff -ruN a/component/isp/common/sensor_list.c b/component/isp/common/sensor_list.c
--- a/component/isp/common/sensor_list.c
+++ b/component/isp/common/sensor_list.c
@@ -1,220 +1,222 @@
-#include "sensor_list.h"
-
-
-const char *snsr_type_name[SAMPLE_SNS_TYPE_BUTT] = {
-	/* ------ LINEAR BEGIN ------*/
-	"BRIGATES_BG0808_MIPI_2M_30FPS_10BIT",
-	"BYD_BF2253L_MIPI_1200P_30FPS_10BIT",
-	"CHIPUP_XS9922B_MODE_720P_1CH_8BIT",
-	"CHIPUP_XS9922B_MODE_720P_2CH_8BIT",
-	"CHIPUP_XS9922B_MODE_720P_3CH_8BIT",
-	"CVSENS_CV2002_MIPI_2M_1080P_30FPS_10BIT",
-	"CVSENS_CV2003_MIPI_2M_1080P_30FPS_10BIT",
-	"CVSENS_CV2003_1L_MIPI_2M_1080P_30FPS_10BIT",
-	"CVSENS_CV2003_1L_SLAVE_MIPI_2M_1080P_30FPS_10BIT",
-	"CVSENS_CV2003_1L_SLAVE1_MIPI_2M_1080P_30FPS_10BIT",
-	"CVSENS_CV4001_MIPI_4M_1440P_25FPS_12BIT",
-	"CVSENS_CV4002_MIPI_4M_1440P_25FPS_10BIT",
-	"CVSENS_CV4002_MIPI_1M_720P_30FPS_10BIT",
-	"GCORE_GC02M1_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC0308_MIPI_480P_30FPS_8BIT",
-	"GCORE_GC0312_MIPI_480P_20FPS_8BIT",
-	"GCORE_GC0329_MIPI_480P_10FPS_8BIT",
-	"GCORE_GC05A2_MIPI_5M_30FPS_10BIT",
-	"GCORE_GC1054_MIPI_1M_30FPS_10BIT",
-	"GCORE_GC1084_MIPI_1M_30FPS_10BIT",
-	"GCORE_GC1084_SLAVE_MIPI_1M_30FPS_10BIT",
-	"GCORE_GC1084_SLAVE1_MIPI_1M_30FPS_10BIT",
-	"GCORE_GC1084_SLAVE2_MIPI_1M_30FPS_10BIT",
-	"GCORE_GC2053_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2053_SLAVE_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2053_1L_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2083_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2093_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2093_MIPI_2M_60FPS_10BIT",
-	"GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC2145_MIPI_2M_12FPS_8BIT",
-	"GCORE_GC2385_1L_MIPI_2M_30FPS_10BIT",
-	"GCORE_GC4023_MIPI_4M_30FPS_10BIT",
-	"GCORE_GC4653_MIPI_4M_30FPS_10BIT",
-	"GCORE_GC4653_SLAVE_MIPI_4M_30FPS_10BIT",
-	"IMGDS_MIS2008_MIPI_2M_1080P_30FPS_12BIT",
-	"NEXTCHIP_N5_1M_2CH_25FPS_8BIT",
-	"NEXTCHIP_N5_2M_25FPS_8BIT",
-	"NEXTCHIP_N6_2M_4CH_25FPS_8BIT",
-	"OV_OS02D10_MIPI_2M_30FPS_10BIT",
-	"OV_OS02D10_SLAVE_MIPI_2M_30FPS_10BIT",
-	"OV_OS02N10_1L_MIPI_2M_15FPS_10BIT",
-	"OV_OS02K10_SLAVE_MIPI_2M_30FPS_12BIT",
-	"OV_OS04A10_MIPI_4M_1440P_30FPS_12BIT",
-	"OV_OS04C10_MIPI_4M_30FPS_12BIT",
-	"OV_OS04C10_MIPI_4M_1440P_30FPS_12BIT",
-	"OV_OS04C10_SLAVE_MIPI_4M_30FPS_12BIT",
-	"OV_OS08A20_MIPI_4M_30FPS_10BIT",
-	"OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT",
-	"OV_OS08A20_MIPI_5M_30FPS_10BIT",
-	"OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT",
-	"OV_OS08A20_MIPI_8M_30FPS_10BIT",
-	"OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT",
-	"OV_OV4689_MIPI_4M_30FPS_10BIT",
-	"OV_OV5647_MIPI_2M_30FPS_10BIT",
-	"OV_OV6211_MIPI_400P_120FPS_10BIT",
-	"OV_OV7251_MIPI_480P_120FPS_10BIT",
-	"PICO384_THERMAL_384X288",
-	"PICO640_THERMAL_479P",
-	"PIXELPLUS_PR2000_2M_30FPS_8BIT",
-	"PIXELPLUS_PR2020_1M_25FPS_8BIT",
-	"PIXELPLUS_PR2020_1M_30FPS_8BIT",
-	"PIXELPLUS_PR2020_2M_25FPS_8BIT",
-	"PIXELPLUS_PR2020_2M_30FPS_8BIT",
-	"PIXELPLUS_PR2100_2M_25FPS_8BIT",
-	"PIXELPLUS_PR2100_2M_2CH_25FPS_8BIT",
-	"PIXELPLUS_PR2100_2M_4CH_25FPS_8BIT",
-	"SMS_SC035GS_MIPI_480P_120FPS_12BIT",
-	"SMS_SC035GS_1L_MIPI_480P_120FPS_10BIT",
-	"SMS_SC035HGS_MIPI_480P_120FPS_12BIT",
-	"SMS_SC035HGS_1L_MIPI_480P_120FPS_10BIT",
-	"SMS_SC132GS_MIPI_1080P_60FPS_10BIT",
-	"SMS_SC132GS_SLAVE_MIPI_1080P_60FPS_10BIT",
-	"SMS_SC1336_1L_MIPI_1M_30FPS_10BIT",
-	"SMS_SC1336_1L_MIPI_1M_60FPS_10BIT",
-	"SMS_SC1346_1L_MIPI_1M_30FPS_10BIT",
-	"SMS_SC1346_1L_MIPI_1M_60FPS_10BIT",
-	"SMS_SC1346_1L_SLAVE_MIPI_1M_30FPS_10BIT",
-	"SMS_SC1346_1L_SLAVE_MIPI_1M_60FPS_10BIT",
-	"SMS_SC200AI_MIPI_2M_30FPS_10BIT",
-	"SMS_SC200AI_1L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC230AI_2L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC230AI_2L_SLAVE_MIPI_2M_30FPS_10BIT",
-	"SMS_SC301IOT_MIPI_3M_30FPS_10BIT",
-	"SMS_SC401AI_MIPI_3M_30FPS_10BIT",
-	"SMS_SC401AI_MIPI_4M_30FPS_10BIT",
-	"SMS_SC500AI_MIPI_4M_30FPS_10BIT",
-	"SMS_SC500AI_MIPI_5M_30FPS_10BIT",
-	"SMS_SC501AI_2L_MIPI_5M_30FPS_10BIT",
-	"SMS_SC531AI_2L_MIPI_5M_30FPS_10BIT",
-	"SMS_SC850SL_MIPI_8M_30FPS_12BIT",
-	"SMS_SC3332_MIPI_3M_30FPS_10BIT",
-	"SMS_SC3335_MIPI_3M_30FPS_10BIT",
-	"SMS_SC3335_SLAVE_MIPI_3M_30FPS_10BIT",
-	"SMS_SC3336_MIPI_3M_30FPS_10BIT",
-	"SMS_SC3336_1L_MIPI_3M_20FPS_10BIT",
-	"SMS_SC3336P_MIPI_3M_30FPS_10BIT",
-	"SMS_SC2331_1L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2331_1L_SLAVE_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2331_1L_SLAVE1_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2335_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336_SLAVE_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336_SLAVE1_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336_1L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336P_MIPI_2M_30FPS_10BIT",
-	"SMS_SC2336P_1L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC223A_1L_MIPI_2M_30FPS_10BIT",
-	"SMS_SC4210_MIPI_4M_30FPS_12BIT",
-	"SMS_SC4336_MIPI_4M_30FPS_10BIT",
-	"SMS_SC4336P_MIPI_4M_30FPS_10BIT",
-	"SMS_SC4336P_1L_MIPI_4M_20FPS_10BIT",
-	"SMS_SC4336P1_1L_MIPI_4M_20FPS_10BIT",
-	"SMS_SC5336_2L_MIPI_5M_30FPS_10BIT",
-	"SMS_SC8238_MIPI_8M_30FPS_10BIT",
-	"SOI_F23_MIPI_2M_30FPS_10BIT",
-	"SOI_F35_MIPI_2M_30FPS_10BIT",
-	"SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT",
-	"SOI_F37P_MIPI_2M_30FPS_10BIT",
-	"SOI_H65_MIPI_1M_30FPS_10BIT",
-	"SOI_K06_MIPI_4M_25FPS_10BIT",
-	"SOI_Q03_MIPI_3M_30FPS_10BIT",
-	"SOI_Q03P_MIPI_3M_30FPS_10BIT",
-	"SOI_F53_MIPI_2M_30FPS_10BIT",
-	"SOI_F352_MIPI_2M_30FPS_10BIT",
-	"SOI_K306_MIPI_4M_25FPS_10BIT",
-	"SONY_IMX290_MIPI_1M_30FPS_12BIT",
-	"SONY_IMX290_MIPI_2M_60FPS_12BIT",
-	"SONY_IMX307_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX307_2L_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX307_SUBLVDS_2M_30FPS_12BIT",
-	"SONY_IMX307_MIPI_2M_60FPS_12BIT",
-	"SONY_IMX307_SUBLVDS_2M_60FPS_12BIT",
-#ifdef FPGA_PORTING
-	"SONY_IMX327_MIPI_1M_30FPS_10BIT",
-#endif
-	"SONY_IMX327_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX327_2L_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX327_SUBLVDS_2M_30FPS_12BIT",
-	"SONY_IMX327_MIPI_2M_60FPS_12BIT",
-	"SONY_IMX334_MIPI_8M_30FPS_12BIT",
-	"SONY_IMX335_MIPI_4M_30FPS_12BIT",
-	"SONY_IMX335_MIPI_4M_1600P_30FPS_12BIT",
-	"SONY_IMX335_2L_MIPI_4M_30FPS_10BIT",
-	"SONY_IMX335_MIPI_5M_30FPS_12BIT",
-	"SONY_IMX335_MIPI_2M_60FPS_10BIT",
-	"SONY_IMX335_MIPI_4M_60FPS_10BIT",
-	"SONY_IMX335_MIPI_5M_60FPS_10BIT",
-	"SONY_IMX347_MIPI_4M_60FPS_12BIT",
-	"SONY_IMX385_MIPI_2M_30FPS_12BIT",
-	"SONY_IMX675_MIPI_5M_30FPS_12BIT",
-	"TECHPOINT_TP2850_MIPI_2M_30FPS_8BIT",
-	"TECHPOINT_TP2850_MIPI_4M_30FPS_8BIT",
-	"TECHPOINT_TP2825_MIPI_2M_30FPS_8BIT",
-	"TECHPOINT_TP2863_MIPI_1M_25FPS_8BIT",
-	"TECHPOINT_TP2863_MIPI_2M_25FPS_8BIT",
-	"TECHPOINT_TP2863_MIPI_1M_30FPS_8BIT",
-	"TECHPOINT_TP2863_MIPI_2M_30FPS_8BIT",
-	"VIVO_MCS369_2M_30FPS_12BIT",
-	"VIVO_MCS369Q_4M_30FPS_12BIT",
-	"VIVO_MM308M2_2M_25FPS_8BIT",
-	"LONTIUM_LT6911_2M_60FPS_8BIT",
-	"LONTIUM_LT7911_2M_60FPS_8BIT",
-	/* ------ LINEAR END ------*/
-
-	/* ------ WDR 2TO1 BEGIN ------*/
-	"BRIGATES_BG0808_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"CVSENS_CV2002_MIPI_2M_1080P_30FPS_WDR2TO1",
-	"CVSENS_CV4001_MIPI_4M_1440P_15FPS_WDR2TO1",
-	"GCORE_GC2093_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"OV_OS04A10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1",
-	"OV_OS04C10_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"OV_OS04C10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1",
-	"OV_OS04C10_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_MIPI_5M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_MIPI_8M_30FPS_10BIT_WDR2TO1",
-	"OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC1336_1L_MIPI_1M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC1336_1L_MIPI_1M_60FPS_10BIT_WDR2TO1",
-	"SMS_SC1346_1L_MIPI_1M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC1346_1L_MIPI_1M_60FPS_10BIT_WDR2TO1",
-	"SMS_SC200AI_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC500AI_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC500AI_MIPI_5M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC850SL_MIPI_8M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC4210_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"SMS_SC8238_MIPI_8M_15FPS_10BIT_WDR2TO1",
-	"SOI_F35_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"SOI_F352_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"SONY_IMX307_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX307_2L_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX307_SUBLVDS_2M_30FPS_12BIT_WDR2TO1",
-#ifdef FPGA_PORTING
-	"SONY_IMX327_MIPI_1M_30FPS_10BIT_WDR2TO1",
-#endif
-	"SONY_IMX327_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX327_2L_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX327_SUBLVDS_2M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX334_MIPI_8M_30FPS_12BIT_WDR2TO1",
-	"SONY_IMX335_MIPI_2M_30FPS_10BIT_WDR2TO1",
-	"SONY_IMX335_MIPI_4M_30FPS_10BIT_WDR2TO1",
-	"SONY_IMX335_MIPI_4M_1600P_30FPS_10BIT_WDR2TO1",
-	"SONY_IMX335_MIPI_5M_30FPS_10BIT_WDR2TO1",
-	"SONY_IMX675_MIPI_5M_25FPS_12BIT_WDR2TO1",
-	/* ------ WDR 2TO1 END ------*/
-};
+#include "sensor_list.h"
+
+
+const char *snsr_type_name[SAMPLE_SNS_TYPE_BUTT] = {
+	/* ------ LINEAR BEGIN ------*/
+	"BRIGATES_BG0808_MIPI_2M_30FPS_10BIT",
+	"BYD_BF2253L_MIPI_1200P_30FPS_10BIT",
+	"CHIPUP_XS9922B_MODE_720P_1CH_8BIT",
+	"CHIPUP_XS9922B_MODE_720P_2CH_8BIT",
+	"CHIPUP_XS9922B_MODE_720P_3CH_8BIT",
+	"CVSENS_CV2002_MIPI_2M_1080P_30FPS_10BIT",
+	"CVSENS_CV2003_MIPI_2M_1080P_30FPS_10BIT",
+	"CVSENS_CV2003_1L_MIPI_2M_1080P_30FPS_10BIT",
+	"CVSENS_CV2003_1L_SLAVE_MIPI_2M_1080P_30FPS_10BIT",
+	"CVSENS_CV2003_1L_SLAVE1_MIPI_2M_1080P_30FPS_10BIT",
+	"CVSENS_CV4001_MIPI_4M_1440P_25FPS_12BIT",
+	"CVSENS_CV4002_MIPI_4M_1440P_25FPS_10BIT",
+	"CVSENS_CV4002_MIPI_1M_720P_30FPS_10BIT",
+	"GCORE_GC02M1_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC0308_MIPI_480P_30FPS_8BIT",
+	"GCORE_GC0312_MIPI_480P_20FPS_8BIT",
+	"GCORE_GC0329_MIPI_480P_10FPS_8BIT",
+	"GCORE_GC05A2_MIPI_5M_30FPS_10BIT",
+	"GCORE_GC1054_MIPI_1M_30FPS_10BIT",
+	"GCORE_GC1084_MIPI_1M_30FPS_10BIT",
+	"GCORE_GC1084_SLAVE_MIPI_1M_30FPS_10BIT",
+	"GCORE_GC1084_SLAVE1_MIPI_1M_30FPS_10BIT",
+	"GCORE_GC1084_SLAVE2_MIPI_1M_30FPS_10BIT",
+	"GCORE_GC2053_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2053_SLAVE_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2053_1L_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2083_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2093_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2093_MIPI_2M_60FPS_10BIT",
+	"GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC2145_MIPI_2M_12FPS_8BIT",
+	"GCORE_GC2385_1L_MIPI_2M_30FPS_10BIT",
+	"GCORE_GC4023_MIPI_4M_30FPS_10BIT",
+	"GCORE_GC4653_MIPI_4M_30FPS_10BIT",
+	"GCORE_GC4653_SLAVE_MIPI_4M_30FPS_10BIT",
+	"IMGDS_MIS2008_MIPI_2M_1080P_30FPS_12BIT",
+	"NEXTCHIP_N5_1M_2CH_25FPS_8BIT",
+	"NEXTCHIP_N5_2M_25FPS_8BIT",
+	"NEXTCHIP_N6_2M_4CH_25FPS_8BIT",
+	"OV_OS02D10_MIPI_2M_30FPS_10BIT",
+	"OV_OS02D10_SLAVE_MIPI_2M_30FPS_10BIT",
+	"OV_OS02N10_1L_MIPI_2M_15FPS_10BIT",
+	"OV_OS02K10_SLAVE_MIPI_2M_30FPS_12BIT",
+	"OV_OS04A10_MIPI_4M_1440P_30FPS_12BIT",
+	"OV_OS04C10_MIPI_4M_30FPS_12BIT",
+	"OV_OS04C10_MIPI_4M_1440P_30FPS_12BIT",
+	"OV_OS04C10_SLAVE_MIPI_4M_30FPS_12BIT",
+	"OV_OS08A20_MIPI_4M_30FPS_10BIT",
+	"OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT",
+	"OV_OS08A20_MIPI_5M_30FPS_10BIT",
+	"OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT",
+	"OV_OS08A20_MIPI_8M_30FPS_10BIT",
+	"OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT",
+	"OV_OV4689_MIPI_4M_30FPS_10BIT",
+	"OV_OV5647_MIPI_2M_30FPS_10BIT",
+	"OV_OV5647_MIPI_1M2_30FPS_10BIT",
+	"OV_OV5647_MIPI_480P_30FPS_10BIT",
+	"OV_OV6211_MIPI_400P_120FPS_10BIT",
+	"OV_OV7251_MIPI_480P_120FPS_10BIT",
+	"PICO384_THERMAL_384X288",
+	"PICO640_THERMAL_479P",
+	"PIXELPLUS_PR2000_2M_30FPS_8BIT",
+	"PIXELPLUS_PR2020_1M_25FPS_8BIT",
+	"PIXELPLUS_PR2020_1M_30FPS_8BIT",
+	"PIXELPLUS_PR2020_2M_25FPS_8BIT",
+	"PIXELPLUS_PR2020_2M_30FPS_8BIT",
+	"PIXELPLUS_PR2100_2M_25FPS_8BIT",
+	"PIXELPLUS_PR2100_2M_2CH_25FPS_8BIT",
+	"PIXELPLUS_PR2100_2M_4CH_25FPS_8BIT",
+	"SMS_SC035GS_MIPI_480P_120FPS_12BIT",
+	"SMS_SC035GS_1L_MIPI_480P_120FPS_10BIT",
+	"SMS_SC035HGS_MIPI_480P_120FPS_12BIT",
+	"SMS_SC035HGS_1L_MIPI_480P_120FPS_10BIT",
+	"SMS_SC132GS_MIPI_1080P_60FPS_10BIT",
+	"SMS_SC132GS_SLAVE_MIPI_1080P_60FPS_10BIT",
+	"SMS_SC1336_1L_MIPI_1M_30FPS_10BIT",
+	"SMS_SC1336_1L_MIPI_1M_60FPS_10BIT",
+	"SMS_SC1346_1L_MIPI_1M_30FPS_10BIT",
+	"SMS_SC1346_1L_MIPI_1M_60FPS_10BIT",
+	"SMS_SC1346_1L_SLAVE_MIPI_1M_30FPS_10BIT",
+	"SMS_SC1346_1L_SLAVE_MIPI_1M_60FPS_10BIT",
+	"SMS_SC200AI_MIPI_2M_30FPS_10BIT",
+	"SMS_SC200AI_1L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC230AI_2L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC230AI_2L_SLAVE_MIPI_2M_30FPS_10BIT",
+	"SMS_SC301IOT_MIPI_3M_30FPS_10BIT",
+	"SMS_SC401AI_MIPI_3M_30FPS_10BIT",
+	"SMS_SC401AI_MIPI_4M_30FPS_10BIT",
+	"SMS_SC500AI_MIPI_4M_30FPS_10BIT",
+	"SMS_SC500AI_MIPI_5M_30FPS_10BIT",
+	"SMS_SC501AI_2L_MIPI_5M_30FPS_10BIT",
+	"SMS_SC531AI_2L_MIPI_5M_30FPS_10BIT",
+	"SMS_SC850SL_MIPI_8M_30FPS_12BIT",
+	"SMS_SC3332_MIPI_3M_30FPS_10BIT",
+	"SMS_SC3335_MIPI_3M_30FPS_10BIT",
+	"SMS_SC3335_SLAVE_MIPI_3M_30FPS_10BIT",
+	"SMS_SC3336_MIPI_3M_30FPS_10BIT",
+	"SMS_SC3336_1L_MIPI_3M_20FPS_10BIT",
+	"SMS_SC3336P_MIPI_3M_30FPS_10BIT",
+	"SMS_SC2331_1L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2331_1L_SLAVE_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2331_1L_SLAVE1_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2335_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336_SLAVE_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336_SLAVE1_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336_1L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336P_MIPI_2M_30FPS_10BIT",
+	"SMS_SC2336P_1L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC223A_1L_MIPI_2M_30FPS_10BIT",
+	"SMS_SC4210_MIPI_4M_30FPS_12BIT",
+	"SMS_SC4336_MIPI_4M_30FPS_10BIT",
+	"SMS_SC4336P_MIPI_4M_30FPS_10BIT",
+	"SMS_SC4336P_1L_MIPI_4M_20FPS_10BIT",
+	"SMS_SC4336P1_1L_MIPI_4M_20FPS_10BIT",
+	"SMS_SC5336_2L_MIPI_5M_30FPS_10BIT",
+	"SMS_SC8238_MIPI_8M_30FPS_10BIT",
+	"SOI_F23_MIPI_2M_30FPS_10BIT",
+	"SOI_F35_MIPI_2M_30FPS_10BIT",
+	"SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT",
+	"SOI_F37P_MIPI_2M_30FPS_10BIT",
+	"SOI_H65_MIPI_1M_30FPS_10BIT",
+	"SOI_K06_MIPI_4M_25FPS_10BIT",
+	"SOI_Q03_MIPI_3M_30FPS_10BIT",
+	"SOI_Q03P_MIPI_3M_30FPS_10BIT",
+	"SOI_F53_MIPI_2M_30FPS_10BIT",
+	"SOI_F352_MIPI_2M_30FPS_10BIT",
+	"SOI_K306_MIPI_4M_25FPS_10BIT",
+	"SONY_IMX290_MIPI_1M_30FPS_12BIT",
+	"SONY_IMX290_MIPI_2M_60FPS_12BIT",
+	"SONY_IMX307_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX307_2L_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX307_SUBLVDS_2M_30FPS_12BIT",
+	"SONY_IMX307_MIPI_2M_60FPS_12BIT",
+	"SONY_IMX307_SUBLVDS_2M_60FPS_12BIT",
+#ifdef FPGA_PORTING
+	"SONY_IMX327_MIPI_1M_30FPS_10BIT",
+#endif
+	"SONY_IMX327_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX327_2L_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX327_SUBLVDS_2M_30FPS_12BIT",
+	"SONY_IMX327_MIPI_2M_60FPS_12BIT",
+	"SONY_IMX334_MIPI_8M_30FPS_12BIT",
+	"SONY_IMX335_MIPI_4M_30FPS_12BIT",
+	"SONY_IMX335_MIPI_4M_1600P_30FPS_12BIT",
+	"SONY_IMX335_2L_MIPI_4M_30FPS_10BIT",
+	"SONY_IMX335_MIPI_5M_30FPS_12BIT",
+	"SONY_IMX335_MIPI_2M_60FPS_10BIT",
+	"SONY_IMX335_MIPI_4M_60FPS_10BIT",
+	"SONY_IMX335_MIPI_5M_60FPS_10BIT",
+	"SONY_IMX347_MIPI_4M_60FPS_12BIT",
+	"SONY_IMX385_MIPI_2M_30FPS_12BIT",
+	"SONY_IMX675_MIPI_5M_30FPS_12BIT",
+	"TECHPOINT_TP2850_MIPI_2M_30FPS_8BIT",
+	"TECHPOINT_TP2850_MIPI_4M_30FPS_8BIT",
+	"TECHPOINT_TP2825_MIPI_2M_30FPS_8BIT",
+	"TECHPOINT_TP2863_MIPI_1M_25FPS_8BIT",
+	"TECHPOINT_TP2863_MIPI_2M_25FPS_8BIT",
+	"TECHPOINT_TP2863_MIPI_1M_30FPS_8BIT",
+	"TECHPOINT_TP2863_MIPI_2M_30FPS_8BIT",
+	"VIVO_MCS369_2M_30FPS_12BIT",
+	"VIVO_MCS369Q_4M_30FPS_12BIT",
+	"VIVO_MM308M2_2M_25FPS_8BIT",
+	"LONTIUM_LT6911_2M_60FPS_8BIT",
+	"LONTIUM_LT7911_2M_60FPS_8BIT",
+	/* ------ LINEAR END ------*/
+
+	/* ------ WDR 2TO1 BEGIN ------*/
+	"BRIGATES_BG0808_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"CVSENS_CV2002_MIPI_2M_1080P_30FPS_WDR2TO1",
+	"CVSENS_CV4001_MIPI_4M_1440P_15FPS_WDR2TO1",
+	"GCORE_GC2093_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"OV_OS04A10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1",
+	"OV_OS04C10_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"OV_OS04C10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1",
+	"OV_OS04C10_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_MIPI_5M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_MIPI_8M_30FPS_10BIT_WDR2TO1",
+	"OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC1336_1L_MIPI_1M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC1336_1L_MIPI_1M_60FPS_10BIT_WDR2TO1",
+	"SMS_SC1346_1L_MIPI_1M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC1346_1L_MIPI_1M_60FPS_10BIT_WDR2TO1",
+	"SMS_SC200AI_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC500AI_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC500AI_MIPI_5M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC850SL_MIPI_8M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC4210_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"SMS_SC8238_MIPI_8M_15FPS_10BIT_WDR2TO1",
+	"SOI_F35_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"SOI_F352_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"SONY_IMX307_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX307_2L_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX307_SUBLVDS_2M_30FPS_12BIT_WDR2TO1",
+#ifdef FPGA_PORTING
+	"SONY_IMX327_MIPI_1M_30FPS_10BIT_WDR2TO1",
+#endif
+	"SONY_IMX327_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX327_2L_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX327_SUBLVDS_2M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX334_MIPI_8M_30FPS_12BIT_WDR2TO1",
+	"SONY_IMX335_MIPI_2M_30FPS_10BIT_WDR2TO1",
+	"SONY_IMX335_MIPI_4M_30FPS_10BIT_WDR2TO1",
+	"SONY_IMX335_MIPI_4M_1600P_30FPS_10BIT_WDR2TO1",
+	"SONY_IMX335_MIPI_5M_30FPS_10BIT_WDR2TO1",
+	"SONY_IMX675_MIPI_5M_25FPS_12BIT_WDR2TO1",
+	/* ------ WDR 2TO1 END ------*/
+};
diff -ruN a/component/isp/common/sensor_list.h b/component/isp/common/sensor_list.h
--- a/component/isp/common/sensor_list.h
+++ b/component/isp/common/sensor_list.h
@@ -1,237 +1,239 @@
-/*
- * Copyright (C) Cvitek Co., Ltd. 2019-2020. All rights reserved.
- *
- * File Name: sensor_list.h
- * Description:
- */
-
-#ifndef __SENSOR_LIST_H__
-#define __SENSOR_LIST_H__
-#include "cvi_comm_isp.h"
-
-
-typedef enum _SAMPLE_SNS_TYPE_E {
-	/* ------ LINEAR BEGIN ------*/
-	BRIGATES_BG0808_MIPI_2M_30FPS_10BIT,
-	BYD_BF2253L_MIPI_1200P_30FPS_10BIT,
-	CHIPUP_XS9922B_MODE_720P_1CH_8BIT,
-	CHIPUP_XS9922B_MODE_720P_2CH_8BIT,
-	CHIPUP_XS9922B_MODE_720P_3CH_8BIT,
-	CVSENS_CV2002_MIPI_2M_1080P_30FPS_10BIT,
-	CVSENS_CV2003_MIPI_2M_1080P_30FPS_10BIT,
-	CVSENS_CV2003_1L_MIPI_2M_1080P_30FPS_10BIT,
-	CVSENS_CV2003_1L_SLAVE_MIPI_2M_1080P_30FPS_10BIT,
-	CVSENS_CV2003_1L_SLAVE1_MIPI_2M_1080P_30FPS_10BIT,
-	CVSENS_CV4001_MIPI_4M_1440P_25FPS_12BIT,
-	CVSENS_CV4002_MIPI_4M_1440P_25FPS_10BIT,
-	CVSENS_CV4002_MIPI_1M_720P_30FPS_10BIT,
-	GCORE_GC02M1_MIPI_2M_30FPS_10BIT,
-	GCORE_GC0308_MIPI_480P_30FPS_8BIT,
-	GCORE_GC0312_MIPI_480P_20FPS_8BIT,
-	GCORE_GC0329_MIPI_480P_10FPS_8BIT,
-	GCORE_GC05A2_MIPI_5M_30FPS_10BIT,
-	GCORE_GC1054_MIPI_1M_30FPS_10BIT,
-	GCORE_GC1084_MIPI_1M_30FPS_10BIT,
-	GCORE_GC1084_SLAVE_MIPI_1M_30FPS_10BIT,
-	GCORE_GC1084_SLAVE1_MIPI_1M_30FPS_10BIT,
-	GCORE_GC1084_SLAVE2_MIPI_1M_30FPS_10BIT,
-	GCORE_GC2053_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2053_SLAVE_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2053_1L_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2083_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2093_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2093_MIPI_2M_60FPS_10BIT,
-	GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT,
-	GCORE_GC2145_MIPI_2M_12FPS_8BIT,
-	GCORE_GC2385_1L_MIPI_2M_30FPS_10BIT,
-	GCORE_GC4023_MIPI_4M_30FPS_10BIT,
-	GCORE_GC4653_MIPI_4M_30FPS_10BIT,
-	GCORE_GC4653_SLAVE_MIPI_4M_30FPS_10BIT,
-	IMGDS_MIS2008_MIPI_2M_1080P_30FPS_12BIT,
-	NEXTCHIP_N5_1M_2CH_25FPS_8BIT,
-	NEXTCHIP_N5_2M_25FPS_8BIT,
-	NEXTCHIP_N6_2M_4CH_25FPS_8BIT,
-	OV_OS02D10_MIPI_2M_30FPS_10BIT,
-	OV_OS02D10_SLAVE_MIPI_2M_30FPS_10BIT,
-	OV_OS02N10_1L_MIPI_2M_15FPS_10BIT,
-	OV_OS02K10_SLAVE_MIPI_2M_30FPS_12BIT,
-	OV_OS04A10_MIPI_4M_1440P_30FPS_12BIT,
-	OV_OS04C10_MIPI_4M_30FPS_12BIT,
-	OV_OS04C10_MIPI_4M_1440P_30FPS_12BIT,
-	OV_OS04C10_SLAVE_MIPI_4M_30FPS_12BIT,
-	OV_OS08A20_MIPI_4M_30FPS_10BIT,
-	OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT,
-	OV_OS08A20_MIPI_5M_30FPS_10BIT,
-	OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT,
-	OV_OS08A20_MIPI_8M_30FPS_10BIT,
-	OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT,
-	OV_OV4689_MIPI_4M_30FPS_10BIT,
-	OV_OV5647_MIPI_2M_30FPS_10BIT,
-	OV_OV6211_MIPI_400P_120FPS_10BIT,
-	OV_OV7251_MIPI_480P_120FPS_10BIT,
-	PICO384_THERMAL_384X288,
-	PICO640_THERMAL_479P,
-	PIXELPLUS_PR2000_2M_30FPS_8BIT,
-	PIXELPLUS_PR2020_1M_25FPS_8BIT,
-	PIXELPLUS_PR2020_1M_30FPS_8BIT,
-	PIXELPLUS_PR2020_2M_25FPS_8BIT,
-	PIXELPLUS_PR2020_2M_30FPS_8BIT,
-	PIXELPLUS_PR2100_2M_25FPS_8BIT,
-	PIXELPLUS_PR2100_2M_2CH_25FPS_8BIT,
-	PIXELPLUS_PR2100_2M_4CH_25FPS_8BIT,
-	SMS_SC035GS_MIPI_480P_120FPS_12BIT,
-	SMS_SC035GS_1L_MIPI_480P_120FPS_10BIT,
-	SMS_SC035HGS_MIPI_480P_120FPS_12BIT,
-	SMS_SC035HGS_1L_MIPI_480P_120FPS_10BIT,
-	SMS_SC132GS_MIPI_1080P_60FPS_10BIT,
-	SMS_SC132GS_SLAVE_MIPI_1080P_60FPS_10BIT,
-	SMS_SC1336_1L_MIPI_1M_30FPS_10BIT,
-	SMS_SC1336_1L_MIPI_1M_60FPS_10BIT,
-	SMS_SC1346_1L_MIPI_1M_30FPS_10BIT,
-	SMS_SC1346_1L_MIPI_1M_60FPS_10BIT,
-	SMS_SC1346_1L_SLAVE_MIPI_1M_30FPS_10BIT,
-	SMS_SC1346_1L_SLAVE_MIPI_1M_60FPS_10BIT,
-	SMS_SC200AI_MIPI_2M_30FPS_10BIT,
-	SMS_SC200AI_1L_MIPI_2M_30FPS_10BIT,
-	SMS_SC230AI_2L_MIPI_2M_30FPS_10BIT,
-	SMS_SC230AI_2L_SLAVE_MIPI_2M_30FPS_10BIT,
-	SMS_SC301IOT_MIPI_3M_30FPS_10BIT,
-	SMS_SC401AI_MIPI_3M_30FPS_10BIT,
-	SMS_SC401AI_MIPI_4M_30FPS_10BIT,
-	SMS_SC500AI_MIPI_4M_30FPS_10BIT,
-	SMS_SC500AI_MIPI_5M_30FPS_10BIT,
-	SMS_SC501AI_2L_MIPI_5M_30FPS_10BIT,
-	SMS_SC531AI_2L_MIPI_5M_30FPS_10BIT,
-	SMS_SC850SL_MIPI_8M_30FPS_12BIT,
-	SMS_SC3332_MIPI_3M_30FPS_10BIT,
-	SMS_SC3335_MIPI_3M_30FPS_10BIT,
-	SMS_SC3335_SLAVE_MIPI_3M_30FPS_10BIT,
-	SMS_SC3336_MIPI_3M_30FPS_10BIT,
-	SMS_SC3336_1L_MIPI_3M_20FPS_10BIT,
-	SMS_SC3336P_MIPI_3M_30FPS_10BIT,
-	SMS_SC2331_1L_MIPI_2M_30FPS_10BIT,
-	SMS_SC2331_1L_SLAVE_MIPI_2M_30FPS_10BIT,
-	SMS_SC2331_1L_SLAVE1_MIPI_2M_30FPS_10BIT,
-	SMS_SC2335_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336_SLAVE_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336_SLAVE1_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336_1L_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336P_MIPI_2M_30FPS_10BIT,
-	SMS_SC2336P_1L_MIPI_2M_30FPS_10BIT,
-	SMS_SC223A_1L_MIPI_2M_30FPS_10BIT,
-	SMS_SC4210_MIPI_4M_30FPS_12BIT,
-	SMS_SC4336_MIPI_4M_30FPS_10BIT,
-	SMS_SC4336P_MIPI_4M_30FPS_10BIT,
-	SMS_SC4336P_1L_MIPI_4M_20FPS_10BIT,
-	SMS_SC4336P1_1L_MIPI_4M_20FPS_10BIT,
-	SMS_SC5336_2L_MIPI_5M_30FPS_10BIT,
-	SMS_SC8238_MIPI_8M_30FPS_10BIT,
-	SOI_F23_MIPI_2M_30FPS_10BIT,
-	SOI_F35_MIPI_2M_30FPS_10BIT,
-	SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT,
-	SOI_F37P_MIPI_2M_30FPS_10BIT,
-	SOI_H65_MIPI_1M_30FPS_10BIT,
-	SOI_K06_MIPI_4M_25FPS_10BIT,
-	SOI_Q03_MIPI_3M_30FPS_10BIT,
-	SOI_Q03P_MIPI_3M_30FPS_10BIT,
-	SOI_F53_MIPI_2M_30FPS_10BIT,
-	SOI_F352_MIPI_2M_30FPS_10BIT,
-	SOI_K306_MIPI_4M_25FPS_10BIT,
-	SONY_IMX290_MIPI_1M_30FPS_12BIT,
-	SONY_IMX290_MIPI_2M_60FPS_12BIT,
-	SONY_IMX307_MIPI_2M_30FPS_12BIT,
-	SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT,
-	SONY_IMX307_2L_MIPI_2M_30FPS_12BIT,
-	SONY_IMX307_SUBLVDS_2M_30FPS_12BIT,
-	SONY_IMX307_MIPI_2M_60FPS_12BIT,
-	SONY_IMX307_SUBLVDS_2M_60FPS_12BIT,
-#ifdef FPGA_PORTING
-	SONY_IMX327_MIPI_1M_30FPS_10BIT,
-#endif
-	SONY_IMX327_MIPI_2M_30FPS_12BIT,
-	SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT,
-	SONY_IMX327_2L_MIPI_2M_30FPS_12BIT,
-	SONY_IMX327_SUBLVDS_2M_30FPS_12BIT,
-	SONY_IMX327_MIPI_2M_60FPS_12BIT,
-	SONY_IMX334_MIPI_8M_30FPS_12BIT,
-	SONY_IMX335_MIPI_4M_30FPS_12BIT,
-	SONY_IMX335_MIPI_4M_1600P_30FPS_12BIT,
-	SONY_IMX335_2L_MIPI_4M_30FPS_10BIT,
-	SONY_IMX335_MIPI_5M_30FPS_12BIT,
-	SONY_IMX335_MIPI_2M_60FPS_10BIT,
-	SONY_IMX335_MIPI_4M_60FPS_10BIT,
-	SONY_IMX335_MIPI_5M_60FPS_10BIT,
-	SONY_IMX347_MIPI_4M_60FPS_12BIT,
-	SONY_IMX385_MIPI_2M_30FPS_12BIT,
-	SONY_IMX675_MIPI_5M_30FPS_12BIT,
-	TECHPOINT_TP2850_MIPI_2M_30FPS_8BIT,
-	TECHPOINT_TP2850_MIPI_4M_30FPS_8BIT,
-	TECHPOINT_TP2825_MIPI_2M_30FPS_8BIT,
-	TECHPOINT_TP2863_MIPI_1M_25FPS_8BIT,
-	TECHPOINT_TP2863_MIPI_2M_25FPS_8BIT,
-	TECHPOINT_TP2863_MIPI_1M_30FPS_8BIT,
-	TECHPOINT_TP2863_MIPI_2M_30FPS_8BIT,
-	VIVO_MCS369_2M_30FPS_12BIT,
-	VIVO_MCS369Q_4M_30FPS_12BIT,
-	VIVO_MM308M2_2M_25FPS_8BIT,
-	LONTIUM_LT6911_2M_60FPS_8BIT,
-	LONTIUM_LT7911_2M_60FPS_8BIT,
-	/* ------ LINEAR END ------*/
-	SAMPLE_SNS_TYPE_LINEAR_BUTT,
-
-	/* ------ WDR 2TO1 BEGIN ------*/
-	BRIGATES_BG0808_MIPI_2M_30FPS_10BIT_WDR2TO1 = SAMPLE_SNS_TYPE_LINEAR_BUTT,
-	CVSENS_CV2002_MIPI_2M_1080P_30FPS_WDR2TO1,
-	CVSENS_CV4001_MIPI_4M_1440P_15FPS_WDR2TO1,
-	GCORE_GC2093_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	OV_OS04A10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1,
-	OV_OS04C10_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	OV_OS04C10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1,
-	OV_OS04C10_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_MIPI_5M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_MIPI_8M_30FPS_10BIT_WDR2TO1,
-	OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT_WDR2TO1,
-	SMS_SC1336_1L_MIPI_1M_30FPS_10BIT_WDR2TO1,
-	SMS_SC1336_1L_MIPI_1M_60FPS_10BIT_WDR2TO1,
-	SMS_SC1346_1L_MIPI_1M_30FPS_10BIT_WDR2TO1,
-	SMS_SC1346_1L_MIPI_1M_60FPS_10BIT_WDR2TO1,
-	SMS_SC200AI_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	SMS_SC500AI_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	SMS_SC500AI_MIPI_5M_30FPS_10BIT_WDR2TO1,
-	SMS_SC850SL_MIPI_8M_30FPS_10BIT_WDR2TO1,
-	SMS_SC4210_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	SMS_SC8238_MIPI_8M_15FPS_10BIT_WDR2TO1,
-	SOI_F35_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	SOI_F352_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	SONY_IMX307_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX307_2L_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX307_SUBLVDS_2M_30FPS_12BIT_WDR2TO1,
-#ifdef FPGA_PORTING
-	SONY_IMX327_MIPI_1M_30FPS_10BIT_WDR2TO1,
-#endif
-	SONY_IMX327_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX327_2L_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX327_SUBLVDS_2M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX334_MIPI_8M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX335_MIPI_2M_30FPS_10BIT_WDR2TO1,
-	SONY_IMX335_MIPI_4M_30FPS_10BIT_WDR2TO1,
-	SONY_IMX335_MIPI_4M_1600P_30FPS_10BIT_WDR2TO1,
-	SONY_IMX335_MIPI_5M_30FPS_10BIT_WDR2TO1,
-	SONY_IMX675_MIPI_5M_25FPS_12BIT_WDR2TO1,
-	SONY_IMX347_MIPI_4M_30FPS_12BIT_WDR2TO1,
-	SONY_IMX385_MIPI_2M_30FPS_12BIT_WDR2TO1,
-	/* ------ WDR 2TO1 END ------*/
-	SAMPLE_SNS_TYPE_BUTT,
-} SAMPLE_SNS_TYPE_E;
-
-extern const char *snsr_type_name[SAMPLE_SNS_TYPE_BUTT];
-
+/*
+ * Copyright (C) Cvitek Co., Ltd. 2019-2020. All rights reserved.
+ *
+ * File Name: sensor_list.h
+ * Description:
+ */
+
+#ifndef __SENSOR_LIST_H__
+#define __SENSOR_LIST_H__
+#include "cvi_comm_isp.h"
+
+
+typedef enum _SAMPLE_SNS_TYPE_E {
+	/* ------ LINEAR BEGIN ------*/
+	BRIGATES_BG0808_MIPI_2M_30FPS_10BIT,
+	BYD_BF2253L_MIPI_1200P_30FPS_10BIT,
+	CHIPUP_XS9922B_MODE_720P_1CH_8BIT,
+	CHIPUP_XS9922B_MODE_720P_2CH_8BIT,
+	CHIPUP_XS9922B_MODE_720P_3CH_8BIT,
+	CVSENS_CV2002_MIPI_2M_1080P_30FPS_10BIT,
+	CVSENS_CV2003_MIPI_2M_1080P_30FPS_10BIT,
+	CVSENS_CV2003_1L_MIPI_2M_1080P_30FPS_10BIT,
+	CVSENS_CV2003_1L_SLAVE_MIPI_2M_1080P_30FPS_10BIT,
+	CVSENS_CV2003_1L_SLAVE1_MIPI_2M_1080P_30FPS_10BIT,
+	CVSENS_CV4001_MIPI_4M_1440P_25FPS_12BIT,
+	CVSENS_CV4002_MIPI_4M_1440P_25FPS_10BIT,
+	CVSENS_CV4002_MIPI_1M_720P_30FPS_10BIT,
+	GCORE_GC02M1_MIPI_2M_30FPS_10BIT,
+	GCORE_GC0308_MIPI_480P_30FPS_8BIT,
+	GCORE_GC0312_MIPI_480P_20FPS_8BIT,
+	GCORE_GC0329_MIPI_480P_10FPS_8BIT,
+	GCORE_GC05A2_MIPI_5M_30FPS_10BIT,
+	GCORE_GC1054_MIPI_1M_30FPS_10BIT,
+	GCORE_GC1084_MIPI_1M_30FPS_10BIT,
+	GCORE_GC1084_SLAVE_MIPI_1M_30FPS_10BIT,
+	GCORE_GC1084_SLAVE1_MIPI_1M_30FPS_10BIT,
+	GCORE_GC1084_SLAVE2_MIPI_1M_30FPS_10BIT,
+	GCORE_GC2053_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2053_SLAVE_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2053_1L_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2083_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2093_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2093_MIPI_2M_60FPS_10BIT,
+	GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT,
+	GCORE_GC2145_MIPI_2M_12FPS_8BIT,
+	GCORE_GC2385_1L_MIPI_2M_30FPS_10BIT,
+	GCORE_GC4023_MIPI_4M_30FPS_10BIT,
+	GCORE_GC4653_MIPI_4M_30FPS_10BIT,
+	GCORE_GC4653_SLAVE_MIPI_4M_30FPS_10BIT,
+	IMGDS_MIS2008_MIPI_2M_1080P_30FPS_12BIT,
+	NEXTCHIP_N5_1M_2CH_25FPS_8BIT,
+	NEXTCHIP_N5_2M_25FPS_8BIT,
+	NEXTCHIP_N6_2M_4CH_25FPS_8BIT,
+	OV_OS02D10_MIPI_2M_30FPS_10BIT,
+	OV_OS02D10_SLAVE_MIPI_2M_30FPS_10BIT,
+	OV_OS02N10_1L_MIPI_2M_15FPS_10BIT,
+	OV_OS02K10_SLAVE_MIPI_2M_30FPS_12BIT,
+	OV_OS04A10_MIPI_4M_1440P_30FPS_12BIT,
+	OV_OS04C10_MIPI_4M_30FPS_12BIT,
+	OV_OS04C10_MIPI_4M_1440P_30FPS_12BIT,
+	OV_OS04C10_SLAVE_MIPI_4M_30FPS_12BIT,
+	OV_OS08A20_MIPI_4M_30FPS_10BIT,
+	OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT,
+	OV_OS08A20_MIPI_5M_30FPS_10BIT,
+	OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT,
+	OV_OS08A20_MIPI_8M_30FPS_10BIT,
+	OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT,
+	OV_OV4689_MIPI_4M_30FPS_10BIT,
+	OV_OV5647_MIPI_2M_30FPS_10BIT,
+	OV_OV5647_MIPI_1M2_30FPS_10BIT,
+	OV_OV5647_MIPI_480P_30FPS_10BIT,
+	OV_OV6211_MIPI_400P_120FPS_10BIT,
+	OV_OV7251_MIPI_480P_120FPS_10BIT,
+	PICO384_THERMAL_384X288,
+	PICO640_THERMAL_479P,
+	PIXELPLUS_PR2000_2M_30FPS_8BIT,
+	PIXELPLUS_PR2020_1M_25FPS_8BIT,
+	PIXELPLUS_PR2020_1M_30FPS_8BIT,
+	PIXELPLUS_PR2020_2M_25FPS_8BIT,
+	PIXELPLUS_PR2020_2M_30FPS_8BIT,
+	PIXELPLUS_PR2100_2M_25FPS_8BIT,
+	PIXELPLUS_PR2100_2M_2CH_25FPS_8BIT,
+	PIXELPLUS_PR2100_2M_4CH_25FPS_8BIT,
+	SMS_SC035GS_MIPI_480P_120FPS_12BIT,
+	SMS_SC035GS_1L_MIPI_480P_120FPS_10BIT,
+	SMS_SC035HGS_MIPI_480P_120FPS_12BIT,
+	SMS_SC035HGS_1L_MIPI_480P_120FPS_10BIT,
+	SMS_SC132GS_MIPI_1080P_60FPS_10BIT,
+	SMS_SC132GS_SLAVE_MIPI_1080P_60FPS_10BIT,
+	SMS_SC1336_1L_MIPI_1M_30FPS_10BIT,
+	SMS_SC1336_1L_MIPI_1M_60FPS_10BIT,
+	SMS_SC1346_1L_MIPI_1M_30FPS_10BIT,
+	SMS_SC1346_1L_MIPI_1M_60FPS_10BIT,
+	SMS_SC1346_1L_SLAVE_MIPI_1M_30FPS_10BIT,
+	SMS_SC1346_1L_SLAVE_MIPI_1M_60FPS_10BIT,
+	SMS_SC200AI_MIPI_2M_30FPS_10BIT,
+	SMS_SC200AI_1L_MIPI_2M_30FPS_10BIT,
+	SMS_SC230AI_2L_MIPI_2M_30FPS_10BIT,
+	SMS_SC230AI_2L_SLAVE_MIPI_2M_30FPS_10BIT,
+	SMS_SC301IOT_MIPI_3M_30FPS_10BIT,
+	SMS_SC401AI_MIPI_3M_30FPS_10BIT,
+	SMS_SC401AI_MIPI_4M_30FPS_10BIT,
+	SMS_SC500AI_MIPI_4M_30FPS_10BIT,
+	SMS_SC500AI_MIPI_5M_30FPS_10BIT,
+	SMS_SC501AI_2L_MIPI_5M_30FPS_10BIT,
+	SMS_SC531AI_2L_MIPI_5M_30FPS_10BIT,
+	SMS_SC850SL_MIPI_8M_30FPS_12BIT,
+	SMS_SC3332_MIPI_3M_30FPS_10BIT,
+	SMS_SC3335_MIPI_3M_30FPS_10BIT,
+	SMS_SC3335_SLAVE_MIPI_3M_30FPS_10BIT,
+	SMS_SC3336_MIPI_3M_30FPS_10BIT,
+	SMS_SC3336_1L_MIPI_3M_20FPS_10BIT,
+	SMS_SC3336P_MIPI_3M_30FPS_10BIT,
+	SMS_SC2331_1L_MIPI_2M_30FPS_10BIT,
+	SMS_SC2331_1L_SLAVE_MIPI_2M_30FPS_10BIT,
+	SMS_SC2331_1L_SLAVE1_MIPI_2M_30FPS_10BIT,
+	SMS_SC2335_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336_SLAVE_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336_SLAVE1_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336_1L_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336P_MIPI_2M_30FPS_10BIT,
+	SMS_SC2336P_1L_MIPI_2M_30FPS_10BIT,
+	SMS_SC223A_1L_MIPI_2M_30FPS_10BIT,
+	SMS_SC4210_MIPI_4M_30FPS_12BIT,
+	SMS_SC4336_MIPI_4M_30FPS_10BIT,
+	SMS_SC4336P_MIPI_4M_30FPS_10BIT,
+	SMS_SC4336P_1L_MIPI_4M_20FPS_10BIT,
+	SMS_SC4336P1_1L_MIPI_4M_20FPS_10BIT,
+	SMS_SC5336_2L_MIPI_5M_30FPS_10BIT,
+	SMS_SC8238_MIPI_8M_30FPS_10BIT,
+	SOI_F23_MIPI_2M_30FPS_10BIT,
+	SOI_F35_MIPI_2M_30FPS_10BIT,
+	SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT,
+	SOI_F37P_MIPI_2M_30FPS_10BIT,
+	SOI_H65_MIPI_1M_30FPS_10BIT,
+	SOI_K06_MIPI_4M_25FPS_10BIT,
+	SOI_Q03_MIPI_3M_30FPS_10BIT,
+	SOI_Q03P_MIPI_3M_30FPS_10BIT,
+	SOI_F53_MIPI_2M_30FPS_10BIT,
+	SOI_F352_MIPI_2M_30FPS_10BIT,
+	SOI_K306_MIPI_4M_25FPS_10BIT,
+	SONY_IMX290_MIPI_1M_30FPS_12BIT,
+	SONY_IMX290_MIPI_2M_60FPS_12BIT,
+	SONY_IMX307_MIPI_2M_30FPS_12BIT,
+	SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT,
+	SONY_IMX307_2L_MIPI_2M_30FPS_12BIT,
+	SONY_IMX307_SUBLVDS_2M_30FPS_12BIT,
+	SONY_IMX307_MIPI_2M_60FPS_12BIT,
+	SONY_IMX307_SUBLVDS_2M_60FPS_12BIT,
+#ifdef FPGA_PORTING
+	SONY_IMX327_MIPI_1M_30FPS_10BIT,
+#endif
+	SONY_IMX327_MIPI_2M_30FPS_12BIT,
+	SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT,
+	SONY_IMX327_2L_MIPI_2M_30FPS_12BIT,
+	SONY_IMX327_SUBLVDS_2M_30FPS_12BIT,
+	SONY_IMX327_MIPI_2M_60FPS_12BIT,
+	SONY_IMX334_MIPI_8M_30FPS_12BIT,
+	SONY_IMX335_MIPI_4M_30FPS_12BIT,
+	SONY_IMX335_MIPI_4M_1600P_30FPS_12BIT,
+	SONY_IMX335_2L_MIPI_4M_30FPS_10BIT,
+	SONY_IMX335_MIPI_5M_30FPS_12BIT,
+	SONY_IMX335_MIPI_2M_60FPS_10BIT,
+	SONY_IMX335_MIPI_4M_60FPS_10BIT,
+	SONY_IMX335_MIPI_5M_60FPS_10BIT,
+	SONY_IMX347_MIPI_4M_60FPS_12BIT,
+	SONY_IMX385_MIPI_2M_30FPS_12BIT,
+	SONY_IMX675_MIPI_5M_30FPS_12BIT,
+	TECHPOINT_TP2850_MIPI_2M_30FPS_8BIT,
+	TECHPOINT_TP2850_MIPI_4M_30FPS_8BIT,
+	TECHPOINT_TP2825_MIPI_2M_30FPS_8BIT,
+	TECHPOINT_TP2863_MIPI_1M_25FPS_8BIT,
+	TECHPOINT_TP2863_MIPI_2M_25FPS_8BIT,
+	TECHPOINT_TP2863_MIPI_1M_30FPS_8BIT,
+	TECHPOINT_TP2863_MIPI_2M_30FPS_8BIT,
+	VIVO_MCS369_2M_30FPS_12BIT,
+	VIVO_MCS369Q_4M_30FPS_12BIT,
+	VIVO_MM308M2_2M_25FPS_8BIT,
+	LONTIUM_LT6911_2M_60FPS_8BIT,
+	LONTIUM_LT7911_2M_60FPS_8BIT,
+	/* ------ LINEAR END ------*/
+	SAMPLE_SNS_TYPE_LINEAR_BUTT,
+
+	/* ------ WDR 2TO1 BEGIN ------*/
+	BRIGATES_BG0808_MIPI_2M_30FPS_10BIT_WDR2TO1 = SAMPLE_SNS_TYPE_LINEAR_BUTT,
+	CVSENS_CV2002_MIPI_2M_1080P_30FPS_WDR2TO1,
+	CVSENS_CV4001_MIPI_4M_1440P_15FPS_WDR2TO1,
+	GCORE_GC2093_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	GCORE_GC2093_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	OV_OS04A10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1,
+	OV_OS04C10_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	OV_OS04C10_MIPI_4M_1440P_30FPS_10BIT_WDR2TO1,
+	OV_OS04C10_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_SLAVE_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_MIPI_5M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_SLAVE_MIPI_5M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_MIPI_8M_30FPS_10BIT_WDR2TO1,
+	OV_OS08A20_SLAVE_MIPI_8M_30FPS_10BIT_WDR2TO1,
+	SMS_SC1336_1L_MIPI_1M_30FPS_10BIT_WDR2TO1,
+	SMS_SC1336_1L_MIPI_1M_60FPS_10BIT_WDR2TO1,
+	SMS_SC1346_1L_MIPI_1M_30FPS_10BIT_WDR2TO1,
+	SMS_SC1346_1L_MIPI_1M_60FPS_10BIT_WDR2TO1,
+	SMS_SC200AI_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	SMS_SC500AI_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	SMS_SC500AI_MIPI_5M_30FPS_10BIT_WDR2TO1,
+	SMS_SC850SL_MIPI_8M_30FPS_10BIT_WDR2TO1,
+	SMS_SC4210_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	SMS_SC8238_MIPI_8M_15FPS_10BIT_WDR2TO1,
+	SOI_F35_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	SOI_F35_SLAVE_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	SOI_F352_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	SONY_IMX307_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX307_2L_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX307_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX307_SUBLVDS_2M_30FPS_12BIT_WDR2TO1,
+#ifdef FPGA_PORTING
+	SONY_IMX327_MIPI_1M_30FPS_10BIT_WDR2TO1,
+#endif
+	SONY_IMX327_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX327_2L_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX327_SLAVE_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX327_SUBLVDS_2M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX334_MIPI_8M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX335_MIPI_2M_30FPS_10BIT_WDR2TO1,
+	SONY_IMX335_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	SONY_IMX335_MIPI_4M_1600P_30FPS_10BIT_WDR2TO1,
+	SONY_IMX335_MIPI_5M_30FPS_10BIT_WDR2TO1,
+	SONY_IMX675_MIPI_5M_25FPS_12BIT_WDR2TO1,
+	SONY_IMX347_MIPI_4M_30FPS_12BIT_WDR2TO1,
+	SONY_IMX385_MIPI_2M_30FPS_12BIT_WDR2TO1,
+	/* ------ WDR 2TO1 END ------*/
+	SAMPLE_SNS_TYPE_BUTT,
+} SAMPLE_SNS_TYPE_E;
+
+extern const char *snsr_type_name[SAMPLE_SNS_TYPE_BUTT];
+
 #endif
\ No newline at end of file
diff -ruN a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos.c b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos.c
--- a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos.c
+++ b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos.c
@@ -66,6 +66,8 @@
 #define OV5647_VTS_ADDR		0x380E
 
 #define OV5647_RES_IS_1080P(w, h)      ((w) == 1920 && (h) == 1080)
+#define OV5647_RES_IS_1280X960(w, h)   ((w) == 1280 && (h) == 960)
+#define OV5647_RES_IS_640X480(w, h)    ((w) == 640 && (h) == 480)
 
 static CVI_S32 cmos_get_ae_default(VI_PIPE ViPipe, AE_SENSOR_DEFAULT_S *pstAeSnsDft)
 {
@@ -161,6 +163,8 @@
 
 	switch (pstSnsState->u8ImgMode) {
 	case OV5647_MODE_1920X1080P30:
+	case OV5647_MODE_1280X960P30:	/* same HTS/VTS: same fps range */
+	case OV5647_MODE_640X480P30:
 		if ((f32Fps <= f32MaxFps) && (f32Fps >= f32MinFps)) {
 			u32VMAX = u32Vts * f32MaxFps / DIV_0_TO_1_FLOAT(f32Fps);
 		} else {
@@ -466,7 +470,9 @@
 
 	switch (u8Mode) {
 	case WDR_MODE_NONE:
-		pstSnsState->u8ImgMode = OV5647_MODE_1920X1080P30;
+		/* keep a linear mode already chosen by cmos_set_image_mode */
+		if (pstSnsState->u8ImgMode >= OV5647_MODE_LINEAR_NUM)
+			pstSnsState->u8ImgMode = OV5647_MODE_1920X1080P30;
 		pstSnsState->enWDRMode = WDR_MODE_NONE;
 		pstSnsState->u32FLStd = g_astOv5647_mode[pstSnsState->u8ImgMode].u32VtsDef;
 		syslog(LOG_INFO, "linear mode\n");
@@ -594,6 +600,10 @@
 		if (pstSnsState->enWDRMode == WDR_MODE_NONE) {
 			if (OV5647_RES_IS_1080P(pstSensorImageMode->u16Width, pstSensorImageMode->u16Height))
 				u8SensorImageMode = OV5647_MODE_1920X1080P30;
+			else if (OV5647_RES_IS_1280X960(pstSensorImageMode->u16Width, pstSensorImageMode->u16Height))
+				u8SensorImageMode = OV5647_MODE_1280X960P30;
+			else if (OV5647_RES_IS_640X480(pstSensorImageMode->u16Width, pstSensorImageMode->u16Height))
+				u8SensorImageMode = OV5647_MODE_640X480P30;
 			else {
 				CVI_TRACE_SNS(CVI_DBG_ERR, "Not support! Width:%d, Height:%d, Fps:%f, WDRMode:%d\n",
 				       pstSensorImageMode->u16Width,
diff -ruN a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_ex.h b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_ex.h
--- a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_ex.h
+++ b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_ex.h
@@ -25,6 +25,8 @@
 
 typedef enum _OV5647_MODE_E {
 	OV5647_MODE_1920X1080P30 = 0,
+	OV5647_MODE_1280X960P30,	/* 2x2 binned, ~full field of view */
+	OV5647_MODE_640X480P30,		/* binned + skipped, ~full field of view */
 	OV5647_MODE_LINEAR_NUM,
 	OV5647_MODE_NUM
 } OV5647_MODE_E;
diff -ruN a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_param.h b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_param.h
--- a/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_param.h
+++ b/component/isp/sensor/cv181x/ov_ov5647/ov5647_cmos_param.h
@@ -54,6 +54,103 @@
 			.u32Step = 1,
 		},
 	},
+	/*
+	 * Full 2592x1944 array, 2x2 binned to 1296x972, output window 1280x960
+	 * centred in it: the ISP post-raw stalls on a 1296-wide frame (not a
+	 * multiple of 32; measured, F7: CSI overflow, no frame done), 1280 is a
+	 * multiple of 64 like 1920. Timing of the 1080p mode kept on purpose
+	 * (same PLL, HTS 2416, VTS 1104): same line time, fps range and exposure
+	 * limits; VTS 1104 > 960 lines + blanking.
+	 */
+	[OV5647_MODE_1280X960P30] = {
+		.name = "1280x960p30",
+		.astImg[0] = {
+			.stSnsSize = {
+				.u32Width = 1280,
+				.u32Height = 960,
+			},
+			.stWndRect = {
+				.s32X = 0,
+				.s32Y = 0,
+				.u32Width = 1280,
+				.u32Height = 960,
+			},
+			.stMaxSize = {
+				.u32Width = 2592,
+				.u32Height = 1944,
+			},
+		},
+
+		.f32MaxFps = 30,
+		.f32MinFps = 0.711, /* 0x4e2 * 30 / 0xFFFF */
+		.u32HtsDef = 2416,
+		.u32VtsDef = 1104,
+		.stExp[0] = {
+			.u16Min = 4,
+			.u16Max = 1104 - 4,
+			.u16Def = 400,
+			.u16Step = 1,
+		},
+		.stAgain[0] = {
+			.u32Min = 1024,
+			.u32Max = 63448,
+			.u32Def = 1024,
+			.u32Step = 1,
+		},
+		.stDgain[0] = {
+			.u32Min = 1024,
+			.u32Max = 1024,
+			.u32Def = 1024,
+			.u32Step = 1,
+		},
+	},	/*
+	 * Array window 2576x1920 (mainline ov5647.c VGA mode), 2x2 binned and
+	 * skipped (0x3814/0x3815 = 0x35) to 640x480. Mainline runs it at 60 fps
+	 * with another PLL; here the 1080p timing stays (30 fps), as the other
+	 * modes.
+	 */
+	[OV5647_MODE_640X480P30] = {
+		.name = "640x480p30",
+		.astImg[0] = {
+			.stSnsSize = {
+				.u32Width = 640,
+				.u32Height = 480,
+			},
+			.stWndRect = {
+				.s32X = 0,
+				.s32Y = 0,
+				.u32Width = 640,
+				.u32Height = 480,
+			},
+			.stMaxSize = {
+				.u32Width = 2592,
+				.u32Height = 1944,
+			},
+		},
+
+		.f32MaxFps = 30,
+		.f32MinFps = 0.711, /* 0x4e2 * 30 / 0xFFFF */
+		.u32HtsDef = 2416,
+		.u32VtsDef = 1104,
+		.stExp[0] = {
+			.u16Min = 4,
+			.u16Max = 1104 - 4,
+			.u16Def = 400,
+			.u16Step = 1,
+		},
+		.stAgain[0] = {
+			.u32Min = 1024,
+			.u32Max = 63448,
+			.u32Def = 1024,
+			.u32Step = 1,
+		},
+		.stDgain[0] = {
+			.u32Min = 1024,
+			.u32Max = 1024,
+			.u32Def = 1024,
+			.u32Step = 1,
+		},
+	},
 };
 
 static ISP_CMOS_BLACK_LEVEL_S g_stIspBlcCalibratio = {
diff -ruN a/component/isp/sensor/cv181x/ov_ov5647/ov5647_sensor_ctl.c b/component/isp/sensor/cv181x/ov_ov5647/ov5647_sensor_ctl.c
--- a/component/isp/sensor/cv181x/ov_ov5647/ov5647_sensor_ctl.c
+++ b/component/isp/sensor/cv181x/ov_ov5647/ov5647_sensor_ctl.c
@@ -13,6 +13,8 @@
 #include "ov5647_cmos_ex.h"
 
 static void ov5647_linear_1080p30_init(VI_PIPE ViPipe);
+static void ov5647_linear_1280x960p30_init(VI_PIPE ViPipe);
+static void ov5647_linear_640x480p30_init(VI_PIPE ViPipe);
 
 CVI_U8 ov5647_i2c_addr = 0x36;        /* I2C Address of OV5647 */
 const CVI_U32 ov5647_addr_byte = 2;
@@ -224,7 +226,12 @@
 
 	delay_ms(10);
 
-	ov5647_linear_1080p30_init(ViPipe);
+	if (g_pastOv5647[ViPipe]->u8ImgMode == OV5647_MODE_1280X960P30)
+		ov5647_linear_1280x960p30_init(ViPipe);
+	else if (g_pastOv5647[ViPipe]->u8ImgMode == OV5647_MODE_640X480P30)
+		ov5647_linear_640x480p30_init(ViPipe);
+	else
+		ov5647_linear_1080p30_init(ViPipe);
 
 	g_pastOv5647[ViPipe]->bInit = CVI_TRUE;
 }
@@ -336,6 +343,228 @@
 	printf("ViPipe:%d,===OV5647 1080P 30fps 10bit LINE Init OK!\n", ViPipe);
 }
 
+/*
+ * 1280x960 @ 30 fps, 2x2 binned, ~full field of view. The 1080p sequence above
+ * with the window/binning/analog registers of the mainline Linux ov5647.c
+ * "2x2 binned full FOV 10-bit mode" (0x3612, 0x3618, 0x3800-0x380b,
+ * 0x3811/0x3813, 0x3814/0x3815, 0x3820/0x3821; 0x3708/0x3709 left at their
+ * reset values, as there), except the output: 1280x960 (0x3808-0x380b)
+ * centred in the 1296x972 binned image (ISP offsets 0x3811 = 12 + 8,
+ * 0x3813 = 6 + 6), because the ISP needs the width aligned (see the mode
+ * table). 0x3821 = 0x01: h-binning without the mirror bit
+ * mainline sets (0x03), to keep the vendor orientation. PLL, MIPI timing,
+ * HTS/VTS unchanged.
+ */
+static void ov5647_linear_1280x960p30_init(VI_PIPE ViPipe)
+{
+	ov5647_write_register(ViPipe, 0x0100, 0x00);
+	ov5647_write_register(ViPipe, 0x0103, 0x01);
+	ov5647_write_register(ViPipe, 0x3035, 0x11);
+	ov5647_write_register(ViPipe, 0x3036, 0x64);
+	ov5647_write_register(ViPipe, 0x303c, 0x11);
+	ov5647_write_register(ViPipe, 0x3821, 0x01);
+	ov5647_write_register(ViPipe, 0x3820, 0x41);
+	ov5647_write_register(ViPipe, 0x370c, 0x0f);
+	ov5647_write_register(ViPipe, 0x3612, 0x59);
+	ov5647_write_register(ViPipe, 0x3618, 0x00);
+	ov5647_write_register(ViPipe, 0x5000, 0x06);
+	ov5647_write_register(ViPipe, 0x5002, 0x40);
+	ov5647_write_register(ViPipe, 0x5003, 0x08);
+	ov5647_write_register(ViPipe, 0x5a00, 0x08);
+	ov5647_write_register(ViPipe, 0x3000, 0xff);
+	ov5647_write_register(ViPipe, 0x3001, 0xff);
+	ov5647_write_register(ViPipe, 0x3002, 0xff);
+	ov5647_write_register(ViPipe, 0x301d, 0xf0);
+	ov5647_write_register(ViPipe, 0x3503, 0x07);
+	ov5647_write_register(ViPipe, 0x3a18, 0x00);
+	ov5647_write_register(ViPipe, 0x3a19, 0xf8);
+	ov5647_write_register(ViPipe, 0x3c01, 0x80);
+	ov5647_write_register(ViPipe, 0x3b07, 0x0c);
+	ov5647_write_register(ViPipe, 0x380c, 0x09);
+	ov5647_write_register(ViPipe, 0x380d, 0x70);
+	ov5647_write_register(ViPipe, 0x380e, 0x04);
+	ov5647_write_register(ViPipe, 0x380f, 0x50);
+	ov5647_write_register(ViPipe, 0x3814, 0x31);
+	ov5647_write_register(ViPipe, 0x3815, 0x31);
+	ov5647_write_register(ViPipe, 0x3808, 0x05);
+	ov5647_write_register(ViPipe, 0x3809, 0x00);
+	ov5647_write_register(ViPipe, 0x380a, 0x03);
+	ov5647_write_register(ViPipe, 0x380b, 0xc0);
+	ov5647_write_register(ViPipe, 0x3811, 0x14);
+	ov5647_write_register(ViPipe, 0x3813, 0x0c);
+	ov5647_write_register(ViPipe, 0x3800, 0x00);
+	ov5647_write_register(ViPipe, 0x3801, 0x00);
+	ov5647_write_register(ViPipe, 0x3802, 0x00);
+	ov5647_write_register(ViPipe, 0x3803, 0x00);
+	ov5647_write_register(ViPipe, 0x3804, 0x0a);
+	ov5647_write_register(ViPipe, 0x3805, 0x3f);
+	ov5647_write_register(ViPipe, 0x3806, 0x07);
+	ov5647_write_register(ViPipe, 0x3807, 0xa3);
+	ov5647_write_register(ViPipe, 0x3630, 0x2e);
+	ov5647_write_register(ViPipe, 0x3632, 0xe2);
+	ov5647_write_register(ViPipe, 0x3633, 0x23);
+	ov5647_write_register(ViPipe, 0x3634, 0x44);
+	ov5647_write_register(ViPipe, 0x3620, 0x64);
+	ov5647_write_register(ViPipe, 0x3621, 0xe0);
+	ov5647_write_register(ViPipe, 0x3600, 0x37);
+	ov5647_write_register(ViPipe, 0x3704, 0xa0);
+	ov5647_write_register(ViPipe, 0x3703, 0x5a);
+	ov5647_write_register(ViPipe, 0x3715, 0x78);
+	ov5647_write_register(ViPipe, 0x3717, 0x01);
+	ov5647_write_register(ViPipe, 0x3731, 0x02);
+	ov5647_write_register(ViPipe, 0x370b, 0x60);
+	ov5647_write_register(ViPipe, 0x3705, 0x1a);
+	ov5647_write_register(ViPipe, 0x3f05, 0x02);
+	ov5647_write_register(ViPipe, 0x3f06, 0x10);
+	ov5647_write_register(ViPipe, 0x3f01, 0x0a);
+	ov5647_write_register(ViPipe, 0x3a08, 0x01);
+	ov5647_write_register(ViPipe, 0x3a09, 0x4b);
+	ov5647_write_register(ViPipe, 0x3a0a, 0x01);
+	ov5647_write_register(ViPipe, 0x3a0b, 0x13);
+	ov5647_write_register(ViPipe, 0x3a0d, 0x04);
+	ov5647_write_register(ViPipe, 0x3a0e, 0x03);
+	ov5647_write_register(ViPipe, 0x3a0f, 0x58);
+	ov5647_write_register(ViPipe, 0x3a10, 0x50);
+	ov5647_write_register(ViPipe, 0x3a1b, 0x58);
+	ov5647_write_register(ViPipe, 0x3a1e, 0x50);
+	ov5647_write_register(ViPipe, 0x3a11, 0x60);
+	ov5647_write_register(ViPipe, 0x3a1f, 0x28);
+	ov5647_write_register(ViPipe, 0x4001, 0x02);
+	ov5647_write_register(ViPipe, 0x4004, 0x04);
+	ov5647_write_register(ViPipe, 0x4000, 0x09);
+	ov5647_write_register(ViPipe, 0x4050, 0x6e);
+	ov5647_write_register(ViPipe, 0x4051, 0x8f);
+	ov5647_write_register(ViPipe, 0x0100, 0x01);
+	ov5647_write_register(ViPipe, 0x3000, 0x00);
+	ov5647_write_register(ViPipe, 0x3001, 0x00);
+	ov5647_write_register(ViPipe, 0x3002, 0x00);
+	ov5647_write_register(ViPipe, 0x3017, 0xe0);
+	ov5647_write_register(ViPipe, 0x301c, 0xfc);
+	ov5647_write_register(ViPipe, 0x3636, 0x06);
+	ov5647_write_register(ViPipe, 0x3016, 0x08);
+	ov5647_write_register(ViPipe, 0x3827, 0xec);
+	ov5647_write_register(ViPipe, 0x3018, 0x44);
+	ov5647_write_register(ViPipe, 0x3035, 0x21);
+	ov5647_write_register(ViPipe, 0x3106, 0xf5);
+	ov5647_write_register(ViPipe, 0x3034, 0x1a);
+	ov5647_write_register(ViPipe, 0x301c, 0xf8);
+
+	ov5647_default_reg_init(ViPipe);
+
+	delay_ms(100);
+
+	printf("ViPipe:%d,===OV5647 1280x960 (2x2 binned) 30fps 10bit LINE Init OK!\n", ViPipe);
+}
+
+/*
+ * 640x480 @ 30 fps, binned + skipped, ~full field of view. The 1080p sequence
+ * above with the registers of the mainline Linux ov5647.c "10-bit VGA full
+ * FOV" mode that set the window, subsampling and analog front end (0x3612,
+ * 0x3618, 0x3708/0x3709, 0x3800-0x380b, 0x3814/0x3815, 0x3820/0x3821,
+ * 0x4004); its PLL (0x3036, for 60 fps) and MIPI timing are not taken, and
+ * 0x3821 drops the mirror bit as in the 1280x960 mode.
+ */
+static void ov5647_linear_640x480p30_init(VI_PIPE ViPipe)
+{
+	ov5647_write_register(ViPipe, 0x0100, 0x00);
+	ov5647_write_register(ViPipe, 0x0103, 0x01);
+	ov5647_write_register(ViPipe, 0x3035, 0x11);
+	ov5647_write_register(ViPipe, 0x3036, 0x64);
+	ov5647_write_register(ViPipe, 0x303c, 0x11);
+	ov5647_write_register(ViPipe, 0x3821, 0x01);
+	ov5647_write_register(ViPipe, 0x3820, 0x41);
+	ov5647_write_register(ViPipe, 0x370c, 0x0f);
+	ov5647_write_register(ViPipe, 0x3612, 0x59);
+	ov5647_write_register(ViPipe, 0x3618, 0x00);
+	ov5647_write_register(ViPipe, 0x5000, 0x06);
+	ov5647_write_register(ViPipe, 0x5002, 0x40);
+	ov5647_write_register(ViPipe, 0x5003, 0x08);
+	ov5647_write_register(ViPipe, 0x5a00, 0x08);
+	ov5647_write_register(ViPipe, 0x3000, 0xff);
+	ov5647_write_register(ViPipe, 0x3001, 0xff);
+	ov5647_write_register(ViPipe, 0x3002, 0xff);
+	ov5647_write_register(ViPipe, 0x301d, 0xf0);
+	ov5647_write_register(ViPipe, 0x3503, 0x07);
+	ov5647_write_register(ViPipe, 0x3a18, 0x00);
+	ov5647_write_register(ViPipe, 0x3a19, 0xf8);
+	ov5647_write_register(ViPipe, 0x3c01, 0x80);
+	ov5647_write_register(ViPipe, 0x3b07, 0x0c);
+	ov5647_write_register(ViPipe, 0x380c, 0x09);
+	ov5647_write_register(ViPipe, 0x380d, 0x70);
+	ov5647_write_register(ViPipe, 0x380e, 0x04);
+	ov5647_write_register(ViPipe, 0x380f, 0x50);
+	ov5647_write_register(ViPipe, 0x3814, 0x35);
+	ov5647_write_register(ViPipe, 0x3815, 0x35);
+	ov5647_write_register(ViPipe, 0x3708, 0x64);
+	ov5647_write_register(ViPipe, 0x3709, 0x52);
+	ov5647_write_register(ViPipe, 0x3808, 0x02);
+	ov5647_write_register(ViPipe, 0x3809, 0x80);
+	ov5647_write_register(ViPipe, 0x380a, 0x01);
+	ov5647_write_register(ViPipe, 0x380b, 0xe0);
+	ov5647_write_register(ViPipe, 0x3800, 0x00);
+	ov5647_write_register(ViPipe, 0x3801, 0x10);
+	ov5647_write_register(ViPipe, 0x3802, 0x00);
+	ov5647_write_register(ViPipe, 0x3803, 0x00);
+	ov5647_write_register(ViPipe, 0x3804, 0x0a);
+	ov5647_write_register(ViPipe, 0x3805, 0x2f);
+	ov5647_write_register(ViPipe, 0x3806, 0x07);
+	ov5647_write_register(ViPipe, 0x3807, 0x9f);
+	ov5647_write_register(ViPipe, 0x3630, 0x2e);
+	ov5647_write_register(ViPipe, 0x3632, 0xe2);
+	ov5647_write_register(ViPipe, 0x3633, 0x23);
+	ov5647_write_register(ViPipe, 0x3634, 0x44);
+	ov5647_write_register(ViPipe, 0x3620, 0x64);
+	ov5647_write_register(ViPipe, 0x3621, 0xe0);
+	ov5647_write_register(ViPipe, 0x3600, 0x37);
+	ov5647_write_register(ViPipe, 0x3704, 0xa0);
+	ov5647_write_register(ViPipe, 0x3703, 0x5a);
+	ov5647_write_register(ViPipe, 0x3715, 0x78);
+	ov5647_write_register(ViPipe, 0x3717, 0x01);
+	ov5647_write_register(ViPipe, 0x3731, 0x02);
+	ov5647_write_register(ViPipe, 0x370b, 0x60);
+	ov5647_write_register(ViPipe, 0x3705, 0x1a);
+	ov5647_write_register(ViPipe, 0x3f05, 0x02);
+	ov5647_write_register(ViPipe, 0x3f06, 0x10);
+	ov5647_write_register(ViPipe, 0x3f01, 0x0a);
+	ov5647_write_register(ViPipe, 0x3a08, 0x01);
+	ov5647_write_register(ViPipe, 0x3a09, 0x4b);
+	ov5647_write_register(ViPipe, 0x3a0a, 0x01);
+	ov5647_write_register(ViPipe, 0x3a0b, 0x13);
+	ov5647_write_register(ViPipe, 0x3a0d, 0x04);
+	ov5647_write_register(ViPipe, 0x3a0e, 0x03);
+	ov5647_write_register(ViPipe, 0x3a0f, 0x58);
+	ov5647_write_register(ViPipe, 0x3a10, 0x50);
+	ov5647_write_register(ViPipe, 0x3a1b, 0x58);
+	ov5647_write_register(ViPipe, 0x3a1e, 0x50);
+	ov5647_write_register(ViPipe, 0x3a11, 0x60);
+	ov5647_write_register(ViPipe, 0x3a1f, 0x28);
+	ov5647_write_register(ViPipe, 0x4001, 0x02);
+	ov5647_write_register(ViPipe, 0x4004, 0x02);
+	ov5647_write_register(ViPipe, 0x4000, 0x09);
+	ov5647_write_register(ViPipe, 0x4050, 0x6e);
+	ov5647_write_register(ViPipe, 0x4051, 0x8f);
+	ov5647_write_register(ViPipe, 0x0100, 0x01);
+	ov5647_write_register(ViPipe, 0x3000, 0x00);
+	ov5647_write_register(ViPipe, 0x3001, 0x00);
+	ov5647_write_register(ViPipe, 0x3002, 0x00);
+	ov5647_write_register(ViPipe, 0x3017, 0xe0);
+	ov5647_write_register(ViPipe, 0x301c, 0xfc);
+	ov5647_write_register(ViPipe, 0x3636, 0x06);
+	ov5647_write_register(ViPipe, 0x3016, 0x08);
+	ov5647_write_register(ViPipe, 0x3827, 0xec);
+	ov5647_write_register(ViPipe, 0x3018, 0x44);
+	ov5647_write_register(ViPipe, 0x3035, 0x21);
+	ov5647_write_register(ViPipe, 0x3106, 0xf5);
+	ov5647_write_register(ViPipe, 0x3034, 0x1a);
+	ov5647_write_register(ViPipe, 0x301c, 0xf8);
+
+	ov5647_default_reg_init(ViPipe);
+
+	delay_ms(100);
+
+	printf("ViPipe:%d,===OV5647 640x480 (binned + skipped) 30fps 10bit LINE Init OK!\n", ViPipe);
+}
+
 
 
 
diff -ruN a/sample/common/sample_comm.h b/sample/common/sample_comm.h
--- a/sample/common/sample_comm.h
+++ b/sample/common/sample_comm.h
@@ -173,6 +173,7 @@
 	PIC_3840x8640,
 	PIC_7688x1124,
 	PIC_640x480,
+	PIC_1280x960,	/* OV5647 2x2 binned, ~full field of view */
 	PIC_479P, /* 632 * 479 */
 	PIC_400x400,
 	PIC_288P, /* 384 * 288 */
diff -ruN a/sample/common/sample_common_sensor.c b/sample/common/sample_common_sensor.c
--- a/sample/common/sample_common_sensor.c
+++ b/sample/common/sample_common_sensor.c
@@ -378,6 +378,12 @@
 	case PIXELPLUS_PR2100_2M_4CH_25FPS_8BIT:
 		*penSize = PIC_1080P;
 		break;
+	case OV_OV5647_MIPI_1M2_30FPS_10BIT:
+		*penSize = PIC_1280x960;
+		break;
+	case OV_OV5647_MIPI_480P_30FPS_10BIT:
+		*penSize = PIC_640x480;
+		break;
 	default:
 		s32Ret = CVI_FAILURE;
 		break;
@@ -527,6 +533,10 @@
 		pstSize->u32Width = 640;
 		pstSize->u32Height = 480;
 		break;
+	case PIC_1280x960:
+		pstSize->u32Width = 1280;
+		pstSize->u32Height = 960;
+		break;
 	case PIC_479P:  /* 632 * 479 */
 		pstSize->u32Width  = 632;
 		pstSize->u32Height = 479;
@@ -930,6 +940,8 @@
 	case SMS_SC230AI_2L_SLAVE_MIPI_2M_30FPS_10BIT:
 	case SONY_IMX335_MIPI_2M_30FPS_10BIT_WDR2TO1:
 	case SONY_IMX347_MIPI_4M_30FPS_12BIT_WDR2TO1:
+	case OV_OV5647_MIPI_480P_30FPS_10BIT:
+	case OV_OV5647_MIPI_1M2_30FPS_10BIT:
 	case OV_OV5647_MIPI_2M_30FPS_10BIT:
 	case GCORE_GC0308_MIPI_480P_30FPS_8BIT:
 	case GCORE_GC05A2_MIPI_5M_30FPS_10BIT:
@@ -1313,6 +1325,8 @@
 #endif
 #if defined(SENSOR_OV_OV5647)
 	case OV_OV5647_MIPI_2M_30FPS_10BIT:
+	case OV_OV5647_MIPI_1M2_30FPS_10BIT:
+	case OV_OV5647_MIPI_480P_30FPS_10BIT:
 		pSnsObj = &stSnsOv5647_Obj;
 		break;
 #endif
